beacon-chain/db/pruner/pruner.go

```diff
@@ -159,6 +169,24 @@ func (p *Service) prune(slot primitives.Slot) error {
 	return nil
 }
 
+func (p *Service) pruneBatches(pruneUpto primitives.Slot) (int, error) {
+	ctx, cancel := context.WithDeadline(p.ctx, time.Now().Add(defaultPruningWindow))
```

Comment:
```suggestion
	ctx, cancel := context.WithTimeout(p.ctx, defaultPruningWindow)
```

A little easier this way and achieves the same result